packages/database/src/migrations/20210121045801-AddTableDhisOrgUnitMapping-modifies-schema.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = function (db) {
+  return db.createTable('dhis_org_unit_mapping', {
+    columns: {
+      id: { type: 'text', primaryKey: true },
+      code: { type: 'text', notNull: true, unique: true },
+      dhis_id: { type: 'text', notNull: true },
+    },
+  });
```

Comment:
What do you think about a more generic table that could be used for other data service entity config later?
```suggestion
  return db.createTable('data_service_entity', {
    columns: {
      id: { type: 'text', primaryKey: true },
      entity_code: { type: 'text', notNull: true, unique: true },
      config: { type: 'jsonb', notNull: true, default: '{}' },
    },
  });
```

That would fit better with our `data_source` model, and be more extensible. Will we ever use it? Hard to know, but I don't think the abstraction adds too much complexity to the overall mapping workflow - do you?

FYI, we do want to break `data_source` out into `data_element` and `data_group` (or even `data_service_element` and `data_service_element_group` to match `data_service_entity`). Having them in the one table at the moment is awkward. See https://github.com/beyondessential/tupaia-backlog/issues/662

Reply:
I think making that table change isn't really needed, since 
- we don't have services other than dhis we need this for (so don't need a generic config instead of dhis_id)
- we don't need any other attributes against entity we would want to give to data broker

I don't see anything like that coming up on the horizon, as far as I know, so I'm happy to leave the abstraction out.

The breaking of data_source into two tables shouldn't affect this either, not sure how that's related?

Reply:
Those are good points.

The connection breaking `data_source` was just that then we'd have three similar tables, all providing information to data broker about how a given dimension maps from tupaia to the external data service:
- `data_service_element`
- `data_service_element_group`
- `data_service_entity`

But, thinking about it now, that may be jumping the gun. Unlike data elements and data groups, the entity doesn't inform the top level of data broker which service we should use - instead it's information used within a specific data service, so it makes sense to have a more specific table.

Happy to stick with what you've got - but just keen to hear what @kael89 thinks first as he put a lot of work into the data broker system too.

Reply:
While working with Tupaia I've seen DHIS specific conventions being generalised time and time again. It may not happen immediately, but it eventually happens after perhaps 1-3 years.

With that in mind, I would also lean towards a more generic table. Even if we only use it for dhis eventually, the extra abstraction doesn't seem to add too much complexity to make us concerned.

On the other hand, it's certainly not the end of the world if this stays DHIS specific for now. Bottom line: I'd be keen to see a more generic table, but would not complain if a specific one is used eventually.

Reply:
Thanks @kael89. A bit of a push there @IgorNadj, but you've got reason enough to stick with your existing table if you still prefer. Over to you :-)

Reply:
Yeah I'd also consider this as a "here's one suggestion, but up to the author" situation 😃

Reply:
No worries, I have changed the table to the generic version